connexta/eve
# Eve92: Login / Admin feature

#### What does this PR do?
Login:
- Login Icon
- Separated existing Microsoft Login feature from EventComponent to be an independent feature
- Account Info
- After logging in, any customization will be tied to the logged-in user, and stored in the backend. Now everyone can have their own customization.

Admin:
- User's get assigned to be an admin role or not. The list of admin role should be stored in `eve/admin.json` following the specific format written in ReadMe. 

Welcome:
- Small cute dialog display at the beginning for sign-in user for greetting. :)

Grafana:
- Reverted back to username/password from api token.

#### Who is reviewing it?

@vinamartin @alchucam @mcalcote 

@Bdthomson @adimka @samuelechu @djblue @gordocanchola

#### How should this be tested?
`make go`, login with Microsoft account after clicking the account icon on the bottom left. Obtain user's ID info. Pass it as a value with `WALLBOARDADMIN` as a key in .env. 
Test it again with `make go` to see the admin status.
Notice any changes made when sign-in will not be shown when sign-out or sign-in with different account.

#### Screenshots

<img width="1914" alt="1" src="https://user-images.githubusercontent.com/40583917/64719547-52ffb100-d47d-11e9-99a9-972f9baa3167.PNG">
<img width="1915" alt="2" src="https://user-images.githubusercontent.com/40583917/64719548-53984780-d47d-11e9-8100-ef29c3779af2.PNG">
<img width="1914" alt="3" src="https://user-images.githubusercontent.com/40583917/64719552-54c97480-d47d-11e9-82aa-f54fd9078eec.PNG">
<img width="1921" alt="4" src="https://user-images.githubusercontent.com/40583917/64719556-55620b00-d47d-11e9-9367-38e9d3e944bf.PNG">


#### Checklist:

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist.
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
